tests/plugins/test_deepspeed_plugin.py

```diff
@@ -28,6 +28,9 @@ def __init__(self):
     def configure_sharded_model(self) -> None:
         self.linear = torch.nn.Linear(32, 2)
 
+    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        self.configure_sharded_model()
```

Comment:
Any way we could add this to LightningModule directly under the condition of DeepSpeed v3 ?

Reply:
Not cleanly. All LightningModule instances would need to call `super().on_load_checkpoint(checkpoint)` then, right? It'd also be strange since people could try to load from checkpoint outside of training
```
if self.trainer and isinstance(self.trainer.training_type_plugin, DeepSpeedPlugin)
```
while also needing to check if deepspeed is installed

Reply:
I was thinking, we could inspect the lightning module cls and see if `configure_sharded_model` has been overridden. If so then we can call this for the user, which would mean it would work.